sacred/config/custom_containers.py

```diff
@@ -226,6 +226,58 @@ def revelation(self):
         return set()
 
 
+class ReadOnlyContainer:
+    def _readonly(self, *args, **kwargs):
+        raise SacredError(
+            'This {} is read-only!'.format(self.__class__.__name__,),
+            filter_traceback='always'
+        )
+
+
+class ReadOnlyDict(dict, ReadOnlyContainer):
+    """
+    A read-only variant of a `dict`
+    """
+    # Overwrite all methods that can modify a dict
+    update = ReadOnlyContainer._readonly
+    pop = ReadOnlyContainer._readonly
+    popitem = ReadOnlyContainer._readonly
+    clear = ReadOnlyContainer._readonly
+    setdefault = ReadOnlyContainer._readonly
+    __setitem__ = ReadOnlyContainer._readonly
+    __delitem__ = ReadOnlyContainer._readonly
+
+
+class ReadOnlyList(list, ReadOnlyContainer):
+    """
+    A read-only variant of a `list`
+    """
+    pop = ReadOnlyContainer._readonly
+    popitem = ReadOnlyContainer._readonly
+    extend = ReadOnlyContainer._readonly
+    append = ReadOnlyContainer._readonly
+    clear = ReadOnlyContainer._readonly
+    setdefault = ReadOnlyContainer._readonly
+    __setitem__ = ReadOnlyContainer._readonly
+    __delitem__ = ReadOnlyContainer._readonly
+
+
+def make_read_only(o):
```

Comment:
Could you add some more unitests for this. Also testing this on more deeply nested structures.